horace_core/sqw/PixelData/@PixelDataFileBacked/PixelDataFileBacked.m

```diff
@@ -386,6 +417,49 @@ function format_dump_data(obj, pix)
 
     end
 
+    methods(Static)
+        function obj = cat(varargin)
+        % Concatenate the given PixelData objects' pixels. This function performs
+        % a straight-forward data concatenation.
+        %
+        %   >> joined_pix = PixelDataBase.cat(pix_data1, pix_data2);
+        %
+        % Input:
+        % ------
+        %   varargin    A cell array of PixelData objects
+        %
+        % Output:
+        % -------
+        %   obj         A PixelData object containing all the pixels in the inputted
+        %               PixelData objects
+
+            is_ldr = cellfun(@(x) isa(x, 'sqw_file_interface'), varargin);
+            ldr = varargin{is_ldr};
+            varargin = varargin(~is_ldr);
+
+            obj = PixelDataFileBacked();
+
+            obj.num_pixels_ = sum(cellfun(@(x) x.num_pixels, varargin));
+
+            if isempty(ldr)
+                obj = obj.get_new_handle();
+            else
+                obj = obj.get_new_handle(ldr);
+            end
+
+            start_idx = 1;
+            for i = 1:numel(varargin)
+                curr_pix = varargin{i};
+                for page = 1:curr_pix.num_pages
+                    [curr_pix,data] = curr_pix.load_page(page);
+                    obj.format_dump_data(data, start_idx);
+                    start_idx = start_idx + size(data,2);
+                end
+            end
+
```

Comment:
without playing with the code, I do not see where pix_data_averages are recalculated. If they are, then fine. Would do. If not -- I am surprized unit test have not found it

Reply:
I still don't know what you mean by `pix_data_averages` do you have an example?